src/tests/unit/services/ExperianService.test.ts

```diff
@@ -0,0 +1,124 @@
+	"refresh_token" : "123456789123456789",
+};
+
+describe("Experian service", () => {
+	
+	beforeAll(() => {
+		experianServiceTest = new ExperianService(logger, experianBaseUrl, mockDynamoDbClient, experianTokenTableName);
+	});
+
+	describe("#checkExperianToken", () => {
+		it("should return access_token if token is valid and not expired", async () => {
+		  mockDynamoDbClient.send = jest.fn().mockResolvedValue({ Item: mockToken });
+		  const result = await experianServiceTest.checkExperianToken();
+		  expect(result).toBe(mockToken);
+		});
+
+		it("should return undefined if no token found", async () => {
+			mockDynamoDbClient.send = jest.fn().mockResolvedValue({});
+			const result = await experianServiceTest.checkExperianToken();
+			expect(result).toBeUndefined();
+		  });
+
+		  it("should return undefined if token has expired", async () => {
+			const expiredToken = {
+				...mockToken,
+				issued_at: 1728639000000, // 11/10/2024 09:30:00.000
+			};
+			mockDynamoDbClient.send = jest.fn().mockResolvedValue({ expiredToken });
+			const result = await experianServiceTest.checkExperianToken();
+			expect(result).toBeUndefined();
+		  });
+	  });
+
+	describe("#saveExperianToken", () => {
+		it("should save Experian token with Id = 1", async () => {
+			mockDynamoDbClient.send = jest.fn().mockResolvedValue({});
+			await experianServiceTest.saveExperianToken(mockToken);
+			expect(mockDynamoDbClient.send).toHaveBeenCalledWith(expect.objectContaining({
+				input: {
+					Item: {
+						"id": "1",
+						...mockToken,
+					},
+					TableName: experianTokenTableName,
+				},
+			}));
+		});
+
+		it("should handle error when sending message to dynamo", async () => {
+			mockDynamoDbClient.send = jest.fn().mockRejectedValueOnce({});
+			await expect(experianServiceTest.saveExperianToken(mockToken)).rejects.toThrow(expect.objectContaining({
+				statusCode: HttpCodesEnum.SERVER_ERROR,
+				message: "Error storing new Experian token",
+			}));
+		});
+	});
+
+	describe("#generateExperianToken", () => {
+		const clientUsername = "123456";
+		const clientPassword = "12345678";
+		const clientSecret = "Test";
+		const clientId = "clientId";
+
+		const expectedParams = {
+			username: clientUsername,
+			password: clientPassword,
+			client_id: clientId,
+			client_secret: clientSecret,
+		};
+
+		const config: AxiosRequestConfig<any> = {
+			headers: {
+				"Content-Type": "application/json",
+				"X-Correlation-Id": "randomId",
+				"X-User-Domain": "cabinetofficegds.com",
+			},
+		};
+
+		it("Should return a valid access token response if a valid access token already exists", async () => {
+			experianServiceTest.checkExperianToken = jest.fn().mockResolvedValue(mockToken);
+			const data = await experianServiceTest.generateExperianToken(clientUsername, clientPassword, clientId, clientSecret);
+			expect(data).toBe(mockToken);
+		});
+
+		it("Should generate a new access token if a valid access token does not exist", async () => {
```

Comment:
Can we add a test for axios returning a 400 or 500 to make sure it presents as an error

Reply:
Done